nexus/db-model/src/network_interface.rs

```diff
@@ -161,26 +161,11 @@ pub struct ServiceNetworkInterface {
 impl ServiceNetworkInterface {
     /// Generate a suitable [`Name`] for the given Omicron zone ID and kind.
     pub fn name(zone_id: OmicronZoneUuid, zone_kind: ZoneKind) -> Name {
-        // Ideally we'd use `zone_kind.to_string()` here, but that uses
-        // underscores as separators which aren't allowed in `Name`s. We also
-        // preserve some existing naming behavior where NTP external networking
-        // is just called "ntp", not "boundary-ntp".
-        //
-        // Most of these zone kinds do not get external networking and therefore
-        // we don't need to be able to generate names for them, but it's simpler
-        // to give them valid descriptions than worry about error handling here.
-        let prefix = match zone_kind {
-            ZoneKind::BoundaryNtp | ZoneKind::InternalNtp => "ntp",
-            ZoneKind::Clickhouse => "clickhouse",
-            ZoneKind::ClickhouseKeeper => "clickhouse-keeper",
-            ZoneKind::CockroachDb => "cockroach",
-            ZoneKind::Crucible => "crucible",
-            ZoneKind::CruciblePantry => "crucible-pantry",
-            ZoneKind::ExternalDns => "external-dns",
-            ZoneKind::InternalDns => "internal-dns",
-            ZoneKind::Nexus => "nexus",
-            ZoneKind::Oximeter => "oximeter",
-        };
+        // Most of these zone kinds do not get external networking and
+        // therefore we don't need to be able to generate names for them, but
+        // it's simpler to give them valid descriptions than worry about error
+        // handling here.
+        let prefix = zone_kind.name_prefix();
```

Comment:
Good idea!

Reply:
The logic is unchanged here btw -- I just moved the string representation in here.